utils/textmate/Syntaxes/Carbon.plist

```diff
@@ -0,0 +1,181 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN"
+        "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+    <dict>
+        <!-- Section metadata -->
+        <key>name</key>
+        <string>carbon</string>
+        <key>uuid</key>
+        <string>04F7F7E3-D5EE-40A4-9899-53CD34D28276</string>
+        <key>scopeName</key>
+        <string>source.carbon</string>
+        <key>fileTypes</key>
+        <array>
+            <string>carbon</string>
+        </array>
+
+        <!-- Folding markers -->
+        <key>foldingStartMarker</key>
+        <string>\{\s*$</string>
+        <key>foldingStopMarker</key>
+        <string>^\s*\}</string>
+
+        <!-- Active patterns from repository -->
+        <key>patterns</key>
+        <array>
+            <!-- Rule: comments -->
+            <dict>
+                <key>include</key>
+                <string>#comments</string>
+            </dict>
+            <!-- Rule: control flow keywords -->
+            <dict>
+                <key>name</key>
+                <string>keyword.control.carbon</string>
+                <key>match</key>
+                <string>\b(break|case|continue|else|if|for|match|return|then|while)\b</string>
+            </dict>
+            <!-- Rule: other keywords -->
+            <dict>
+                <key>name</key>
+                <string>keyword.other.carbon</string>
+                <key>match</key>
+                <string>\b(abstract|addr|alias|and|api|as|auto|base|choice|class|destructor|extends|external|final|forall|fn|impl|import|interface|let|library|namespace|not|or|package|private|protected|returned|template|var|virtual|where)\b</string>
+            </dict>
+            <!-- Builtin types -->
+            <dict>
+                <key>name</key>
+                <string>storage.type.carbon</string>
+                <key>match</key>
+                <string>\b(bool|Carbon\.Int|Carbon\.UInt|f16|f32|f64|f128|i8|i16|i32|i64|i128|i256|Slice|String|StringView|Type|u8|u16|u32|u64|u128|u256)\b</string>
+            </dict>
+            <!-- Rule: numbers - derived from c.tmbundle -->
```

Comment:
The TextMate code is licensed under GPL 3.0, I'm not sure if that interferes with Apache 2.0-with-LLVM-Exception or not?

Reply:
Please let us get back to you on this. @chandlerc

Reply:
Great, if there are any issues I'll happily spend some time to make a new one

Reply:
Yes, this is an issue. For reference, please don't derive code from sources that don't match our license (Apache 2.0 with LLVM-exception -- this is basically us and llvm-project, so fairly limited group).

If you'd like to continue, remove this file and start from scratch. You can read their documentation, but don't copy code.

Here's an independently development highlight syntax that you should use as a basis:
https://github.com/carbon-language/carbon-lang/blob/trunk/utils/highlightjs/highlightjs_carbon_lang.js#L139

(highlightjs may follow a pretty similar structure overall, making it easy to reuse in a lot of this)